pyproject.toml

```diff
@@ -39,58 +39,57 @@ celery = ">=5.3.6,<5.6.0"
 # Cryptographic recipes and primitives.
 # Not directly used by Nautobot itself, but key to many of Nautobot's dependencies.
 # Added as an explicit dependency to protect against various CVEs in older versions.
-cryptography = "~44.0.3"
+cryptography = "~45.0.6"
 # Fundamental web framework for Nautobot
 Django = "~4.2.23"
 # Adds AJAX capabilities to django-tables2
 django-ajax-tables = "~1.1.1"
 # LDAP Support
-django-auth-ldap = {version = "~5.1.0", optional = true}
+django-auth-ldap = {version = "~5.2.0", optional = true}
 # Base functionality for task scheduling with Celery
-django-celery-beat = "~2.6.0"
+django-celery-beat = "~2.7.0"
 # Provides a database backend & models for Celery task results
-django-celery-results = "~2.5.1"
+django-celery-results = "~2.6.0"
 # Management of app configuration via the Django admin UI
 django-constance = "~4.3.0"
 # Permit cross-domain API requests
-django-cors-headers = "~4.4.0"
+django-cors-headers = "~4.7.0"
 # Store files in the database for background tasks
 django-db-file-storage = "~0.5.6.1"
 # AutoSlugField and shell_plus
-django-extensions = "~3.2.3"
+django-extensions = "~4.1"
 # Advanced query filters
 django-filter = "~25.1"
 # Health check endpoint
-django-health-check = "~3.18.3"
+django-health-check = "~3.20.0"
 # Django Jinja - jinja2 backend for django, used to easily register custom jinja filters and render jinja.
 django-jinja = "~2.11.0"
 # Prometheus metrics for Django
-django-prometheus = "~2.3.1"
+django-prometheus = "~2.4.1"
 # Redis cache for Django used for distributed locking
-django-redis = "~5.4.0"
+django-redis = "~6.0.0"
 # Django silk for request profiling
-django-silk = "~5.3.2"
+django-silk = "~5.4.0"
 # External Storage support (i.e. S3)
-# Pinned temporarily to 1.14.3 due to https://github.com/revsys/django-health-check/issues/434
-django-storages = {version = "1.14.3", optional = true}
+django-storages = {version = "~1.14.6", optional = true}
 # Structured logging for Django
-django-structlog = {version = "^9.0.1", extras = ["celery"]}
+django-structlog = {version = "^9.1.1", extras = ["celery"]}
 # Advanced HTML tables
 django-tables2 = "~2.7.5"
 # Tags
 django-taggit = "~6.1.0"
 # Represent time zones in Django
-django-timezone-field = "~7.0"
+django-timezone-field = "~7.1"
 # Tree database structures based on Common Table Expressions
-django-tree-queries = "~0.19.0"
+django-tree-queries = "~0.20.0"
 # Run production webservers such as uWSGI/gunicorn as a Django management command.
 django-webserver = "~1.2.0"
 # REST API framework
-djangorestframework = "~3.15.2"
+djangorestframework = "~3.16.0"
```

Comment:
3.16.1 was released a couple of days ago, need to bump this probably.

There are some changes in 3.16 regarding the handling of `UniqueTogether` and similar constraints, we should look very closely at those to understand the impact.